plugins/in_emitter/emitter.c

```diff
@@ -40,6 +40,24 @@ struct flb_emitter {
     struct flb_input_instance *ins;     /* input instance */
 };
 
+
+/* 
+ * Internal code that uses the in_emitter may want to register a
+ * pause callback to be invoked when the emitter input's pause is invoked.
+ */ 
+typedef void(flb_in_emitter_pause_cb)(void *data, struct flb_config *config);
+
+
+/* 
+ * Optional callbacks for client code that uses in_emitter
+ * Right now, only a pause callback is implemented
+ */ 
+struct flb_emitter_callbacks {
+    flb_in_emitter_pause_cb *pause;
+
+    void *data;
+};
```

Comment:
I would prefer this struct to not be repeated. Is there something preventing you from # include ing ml.h which also contains this struct?

Reply:
So the reason why I did this is because I'm following the pattern of the existing in_emitter code which doesn't have a .h file as far as I can tell for the emitter and then its re-declared in the filters that use it... I don't really understand this code design pattern either @edsiper @nokute78 

https://github.com/fluent/fluent-bit/blob/master/plugins/filter_rewrite_tag/rewrite_tag.h#L57
https://github.com/fluent/fluent-bit/tree/master/plugins/in_emitter

Reply:
I was thinking maybe add a
```
#include ml.h
```
to `emitter.c`

Would that allow for removing the redundancy?

Reply:
Yeah but then that's weird... because the ml filter code doesn't "own" these structures... in my opinion, there should be an emitter.h somewhere, I don't know why the existing code doesn't have that... I suppose I could just refactor it and hope that @edsiper is cool with that..